lib/cookies.js

```diff
@@ -61,6 +61,22 @@ Cookies.prototype = {
       throw new Error('Cannot send secure cookie over unencrypted connection')
     }
 
+    if ( opts && opts.domain && Array.isArray(opts.domain) && req.headers.host ) {
```

Comment:
hmm... that's true, perhaps allowing an array to set headers to be checked on? exit on exists?

[ 'X-Forwarded-Host', 'domain'], for example.